src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterKeyValuePair.cs

```diff
@@ -41,12 +45,14 @@ internal sealed class JsonKeyValuePairConverter<TKey, TValue> : JsonConverter<Ke
             string propertyName = reader.GetString()!;
```

Comment:
I believe this needs to be nullable, but should be addressed outside this PR. cc @layomia